init.c

```diff
@@ -7,9 +7,12 @@
 #include <ctype.h>
 #include <string.h>
 #include <errno.h>
-#include <sys/ipc.h>
 #include <sys/types.h>
+#ifndef __rtems__
 #include <dlfcn.h>
+#else /* __rtems__ */
+#include "dlfcn.h"
```

Comment:
That's a local include. Where does it come from? Is that using the ./os/windows/posix/include/dlfcn.h?

Reply:
No, it's from 5/arm-rtems5/beagleboneblack/lib/include/dlfcn.h, It appears after building rtems-libbsd with posix support

Reply:
And it's in a local include path?

Reply:
Just had a look: That should be found with <dlfcn.h> too. Otherwise there is some odd bug. What does your error message look like if you don't change that?